src/components/header-bar/command-palette/utils/filter.js

```diff
@@ -0,0 +1,88 @@
+import {
+    ALL_APPS_VIEW,
+    ALL_COMMANDS_VIEW,
+    ALL_SHORTCUTS_VIEW,
+    FILTERABLE_ACTION,
+} from './constants.js'
+
+/**
+ * Copied from here:
+ * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#Escaping
+ */
+function escapeRegExpCharacters(text) {
+    return text.replace(/[/.*+?^${}()|[\]\\]/g, '\\$&')
+}
+
+function removePunctuationMarks(text) {
+    return text.replace(/[.,!;:`"'?\-_\s]/g, '')
+}
+
+function removeAccentMarks(str) {
+    /**
+     * normalisation reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
+     */
+    return str.normalize('NFKD').replace(/[\u0300-\u036f]/g, '')
+}
+
+export function processString(text) {
+    const str = removePunctuationMarks(text)
+    return removeAccentMarks(str)
+}
+
+export const filterItemsArray = (items, filter) => {
+    if (!filter.length) {
+        return items
+    }
+    return items.filter(({ displayName, name }) => {
+        const itemName = displayName || name
+        const formattedItemName = itemName.toLowerCase()
+        const formattedFilter = filter.toLowerCase()
+
+        const escapedFilter = escapeRegExpCharacters(formattedFilter)
+        if (formattedItemName.match(escapedFilter)) {
+            return true
+        }
+
+        const normalisedItemName = processString(formattedItemName)
```

Comment:
In a future PR to improve performance we could pre-normalize all items instead of doing it on-demand here